core/src/main/scala/org/trustedanalytics/sparktk/frame/internal/ops/unflatten/UnflattenColumnsFunctions.scala

```diff
@@ -0,0 +1,65 @@
+package org.trustedanalytics.sparktk.frame.internal.ops.unflatten
+
+import org.trustedanalytics.sparktk.frame.{ Schema, DataTypes, Column }
+import org.trustedanalytics.sparktk.frame.internal.RowWrapper
+import org.apache.commons.lang3.StringUtils
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.Row
+
+/**
+ * Functions used for the UnflattenColumns operation.
+ */
+object UnflattenColumnsFunctions extends Serializable {
+
+  def createTargetSchema(schema: Schema, compositeKeyNames: List[String]): Schema = {
+    val keys = schema.copySubset(compositeKeyNames)
+    val converted = schema.columnsExcept(compositeKeyNames).map(col => Column(col.name, DataTypes.string))
+
+    keys.addColumns(converted)
+  }
+
+  def unflattenRddByCompositeKey(compositeKeyIndex: List[Int],
+                                 initialRdd: RDD[(Vector[Any], Iterable[Row])],
+                                 targetSchema: Schema,
+                                 delimiter: String): RDD[Row] = {
+    val rowWrapper = new RowWrapper(targetSchema)
+    val unflattenRdd = initialRdd.map { case (key, row) => key.toArray ++ unflattenRowsForKey(compositeKeyIndex, row, delimiter) }
+
+    unflattenRdd.map(row => rowWrapper.create(row))
+  }
+
+  private def unflattenRowsForKey(compositeKeyIndex: List[Int], groupedByRows: Iterable[Row], delimiter: String): Array[Any] = {
+
+    val rows = groupedByRows.toList
```

Comment:
perf....  Shouldn't have to create a list out of the Iterable[Row], the algo could instead just iterate over it once.   Please add at least a todo: comment stating as much